src/App/Pages/Accounts/LoginSsoPageViewModel.cs

```diff
@@ -63,9 +65,26 @@ public string OrgIdentifier
 
         public async Task InitAsync()
         {
-            if (string.IsNullOrWhiteSpace(OrgIdentifier))
+            try
             {
-                OrgIdentifier = await _stateService.GetRememberedOrgIdentifierAsync();
+                await _deviceActionService.ShowLoadingAsync(AppResources.Loading);
+                if (await TryClaimedDomainLogin())
+                {
+                    return;
+                }
+
+                if (string.IsNullOrWhiteSpace(OrgIdentifier))
+                {
+                    OrgIdentifier = await _stateService.GetRememberedOrgIdentifierAsync();
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.Exception(ex);
```

Comment:
In case an exception is raised, does it leave the view in a good state?

Reply:
Yes. This is an auto navigation feature, if something goes wrong, it will stay in the screen for the user to manually type the organization identifier.